Automattic/wp-calypso
# Purchases: Standardize name of privacy protection upgrade

This pull request fixes https://github.com/Automattic/wp-calypso/issues/1938 by renaming all occurrences of _private registration_ into _privacy protection_ in order to be consistent and reduce confusion, both for users and for us developers. 

#### Testing instructions

You should check that _private registration_ is never mentioned anywhere in the following flow:

1. Run `git checkout update/privacy-protection` and start your server, or open a [live branch](https://calypso.live/?branch=update/privacy-protection)
2. Open the [`Home` page](http://calypso.localhost:3000/) and log in
3. Sandbox the store
4. Navigate to the [`Domains` page](http://calypso.localhost:3000/domains/add/stephanethomas222.wordpress.com)
5. Purchase a plan with a domain registration and privacy protection
6. Purchase another domain without privacy protection this time
7. Navigate to the `Domains` page again
8. Click the second domain registration
9. Click the `None` flag in front of `Privacy Protection:`
10. Click the `Enable Privacy Protection` link
11. Click the `Add Privacy Protection` button
12. Navigate to the [`Purchases` page](http://calypso.localhost:3000/me/purchases) now
13. Click the domain registration with privacy protection
14. Click the `Cancel Privacy Protection` link
15. Click the `Cancel Privacy Protection` button

#### Additional notes
 
The following occurrences have not been renamed because this would involve updating the API:

* API properties:
[`has_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/lib/domains/assembler.js#L30)
[`has_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/lib/purchases/assembler.js#L26)
[`has_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/state/sites/domains/assembler.js#L18)

* API error codes:
[`enable_private_reg_failed`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/my-sites/upgrades/domain-management/transfer/transfer-out/unlocked.jsx#L41)
[`unlock_domain_and_disable_private_reg_failed` and `unlock_domain_failed`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/my-sites/upgrades/domain-management/transfer/transfer-out/shared.js#L15-L24)

* Tests and fixtures:
[`has_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/state/purchases/test/reducer.js#L205-L217)
[`has_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/state/sites/domains/test/fixture.js#L102-L134)

Finally the [`cancel_private_registration`](https://github.com/Automattic/wp-calypso/blob/015acc7f439ebe21529584b3efea13f635091a56/client/me/purchases/cancel-private-registration/index.jsx#L53-L59) stat has not been renamed because this would break history.

#### Reviews
